### PR TITLE
Add tests for special cases in capture loop

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/111254

### CODE REVIEW

File: t/23-baseclass.t [hunk 1]
```diff
@@ -467,4 +467,55 @@ subtest 'requesting full screen update' => sub {
...
+            push @{$self->{writes}}, \@args;
+    });
+    $baseclass_mock->redefine(check_select_rate => sub ($buckets, @args) {
+            $buckets->{BUCKET}->{$file_no} = 'some count';
+            return 1;    # pretend console is not responding
+    });
+    ok !$baseclass->stall_detected, 'no stall detected so far';
+
+    # actually run the loop
+    eval { $baseclass->do_capture };    # throw_ok does not work with OpenQA::Exception::ConsoleReadError
```
Comment (kalikiana) [inline t/23-baseclass.t:512]: Why doesn't `throw_ok` work with `OpenQA::Exception::ConsoleReadError`? 🤔

File: t/23-baseclass.t [hunk 2]
```diff
@@ -467,4 +467,55 @@ subtest 'requesting full screen update' => sub {
 
 is($baseclass->get_wait_still_screen_on_here_doc_input({}), 0, 'wait_still_screen on here doc is off by default!');
 
+subtest 'corner cases of do_capture' => sub {
+    # note: This test covers a few corner cases of do_capture that are not otherwise covered anyways:
+    #       using external video encoder, stall detection, screen update request, unresponsive console
+
+    # prepare file handle to have something to read from
+    open my $file_fh, '<', "$Bin/23-baseclass.t";    # just open this Perl script itself
```
Comment (kalikiana) [inline t/23-baseclass.t:475]:
```suggestion
    open my $file_fh, '<', "$Bin/" . basename(__FILE__);    # just open this Perl script itself
```

**Review by okurz — APPROVED** [review]

File: t/23-baseclass.t (hunk 1 above)
Comment (Martchus) [reply t/23-baseclass.t:512]: I'm not sure yet. Maybe because it is not a plain string.

**Review by baierjan — APPROVED** [review]
